### PR TITLE
Dash table loading states

### PR DESCRIPTION
**Pull request by shammamah**
## About

* Added a small example for the changes made in https://github.com/plotly/dash-table/pull/484
* Added a temporary tarball of the table for testing purposes, since it has not yet been released (will delete this, the `4.1.0` tarball left over from https://github.com/plotly/dash-docs/pull/479, and the corresponding line in `requirements.txt` upon approval and before merge) 
* Fixed some formatting for the "Editable" page of the table docs
* Changed some prop names that were out-of-date in the existing examples

Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

### CODE REVIEW

File: tutorial/examples/table/editing_columns.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
             'name': 'Column {}'.format(i),
             'id': 'column-{}'.format(i),
             'deletable': True,
-            'editable_name': True
+            'editable': True
```
Comment (Marc-Andre-Rivet) [inline tutorial/examples/table/editing_columns.py]: Good catch. `editable_name` actually became `renamable`.

`renamable` has to do with changing the header text
`editable` has to do with changing the data cells content

File: tutorial/examples/table/editing_columns.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def update_columns(n_clicks, value, existing_columns):
     if n_clicks > 0:
         existing_columns.append({
             'id': value, 'name': value,
-            'editable_name': True, 'deletable': True
+            'editable': True, 'deletable': True
```
Comment (Marc-Andre-Rivet) [inline tutorial/examples/table/editing_columns.py]: idem

File: tutorial/examples/table/editing_rows_and_columns.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def update_columns(n_clicks, value, existing_columns):
     if n_clicks > 0:
         existing_columns.append({
             'id': value, 'name': value,
-            'editable_name': True, 'deletable': True
+            'editable': True, 'deletable': True
```
Comment (Marc-Andre-Rivet) [inline tutorial/examples/table/editing_rows_and_columns.py]: idem

File: tutorial/table/editing_recipes_chapter.py [hunk 4]
```diff
@@ -232,9 +270,9 @@
             columns=[
                 {"name": ["", "Year"], "id": "year", "clearable": "first" },
                 {"name": ["City", "Montreal"], "id": "montreal", "deletable": [False, True]},
-                {"name": ["City", "Toronto"], "id": "toronto", "renameable": True },
+                {"name": ["City", "Toronto"], "id": "toronto", "renamable": True },
```
Comment (Marc-Andre-Rivet) [inline tutorial/table/editing_recipes_chapter.py]: 😅

File: tutorial/table/editing_recipes_chapter.py [hunk 5]
```diff
@@ -28,55 +29,92 @@
...
+    control over whether the table is editable. If the `data` property
+    is loading (e.g., while retrieving data from a server), you will
+    be unable to edit the cells and the dropdowns. This avoids cases
+    in which the edited value (input from the user) conflicts with the
+    value that is returned by the server.
+
+    In the example below, you can use the dropdown to choose to load
+    either the `style_cell` property or the `data` property. When you
+    select the property, there will be a simulated delay (to mimic a
+    delay you might get when communicating with a server).
```
Comment (Marc-Andre-Rivet) [inline tutorial/table/editing_recipes_chapter.py]: I'd add a comment indicating that the `dash-loading` can be used to apply specific styling, cursor, etc. while the table is waiting.

File: tutorial/examples/table/editing_columns.py (hunk 1 above)
Comment (shammamah) [reply tutorial/examples/table/editing_columns.py]: @Marc-Andre-Rivet My mistake -- what does `renameable` become, then?

Comment (Marc-Andre-Rivet) [reply tutorial/examples/table/editing_columns.py]: `renamable` :)

**Pull request merged**